service/worker/scanner/scanner.go

```diff
@@ -104,6 +104,7 @@ type (
 	Scanner struct {
 		context scannerContext
 		wg      sync.WaitGroup
+		cancel  context.CancelFunc
```

Comment:
could call it `lifecycleCancel` or similar to be clear that the lifetime matches the Scanner?

Reply:
Makes sense, changed